orttraining/orttraining/core/optimizer/megatron_transformer.cc

```diff
@@ -653,13 +1179,27 @@ Status MegatronTransformer::ApplyImpl(Graph& graph, bool& modified, int graph_le
         output->ClearShape();
   }
 
+  for (auto x : updated_weight_names_) {
+    auto old_initializer_name = x.first;
+    auto new_initializer_name = x.second;
+    if (weights_to_train_.find(old_initializer_name) != weights_to_train_.end()) {
+      weights_to_train_.erase(old_initializer_name);
```

Comment:
shall we do keep the original order of weights_to_train_?

Reply:
changed